tamagotchi/src/contract.rs

```diff
@@ -0,0 +1,182 @@
+use gear_lib::non_fungible_token::{io::NFTTransfer, nft_core::*, state::*, token::*};
+use gear_lib_derive::{NFTCore, NFTMetaState, NFTStateKeeper};
+use gmeta::Metadata;
+use gstd::{errors::Result as GstdResult, exec, msg, prelude::*, ActorId, MessageId};
+use hashbrown::HashMap;
+use primitive_types::{H256, U256};
+use program_io::{
+    ProgramMetadata,
+    Tamagotchi,
+    TmgAction,
+    TmgEvent
+};
+
+const HUNGER_PER_BLOCK: u64 = 1;
+const ENERGY_PER_BLOCK: u64 = 2;
+const BOREDOM_PER_BLOCK: u64 = 2;
+const FILL_PER_SLEEP: u64 = 1000;
+const FILL_PER_FEED: u64 = 1000;
+const FILL_PER_ENTERTAINMENT: u64 = 1000;
+
+static mut TAMAGOTCHI: Option<Tamagotchi> = None;
+
+fn state_mut() -> &'static mut Tamagotchi {
+    let state = unsafe { TAMAGOTCHI.as_mut() };
+
+    debug_assert!(state.is_some(), "state isn't initialized");
+
+    unsafe { state.unwrap_unchecked() }
+}
+
+fn state_ref() -> &'static Tamagotchi {
+    let state = unsafe { TAMAGOTCHI.as_ref() };
+
+    debug_assert!(state.is_some(), "state isn't initialized");
+
+    unsafe { state.unwrap_unchecked() }
+}
+
+#[no_mangle]
+unsafe extern "C" fn init() {
+    let tamagotchi_name: String = msg::load().expect("Error in init message");
+    let new_tamagotchi: Tamagotchi = Tamagotchi {
+        name: tamagotchi_name,
+        date_of_birth: exec::block_timestamp(),
+        owner: msg::source(),
+        fed: 5000,
+        fed_block: 0,
+        entertained: 5000,
+        entertained_block: 0,
+        rested: 5000,
+        rested_block: 0
+    };
+    TAMAGOTCHI = Some(new_tamagotchi);
+    msg::reply("successful initialization!", 0).expect("error in reply");
+}
+
+#[no_mangle]
+unsafe extern "C" fn handle() {
+    let type_message: TmgAction = msg::load().expect("error in load message");
+    let tamagotchi = state_mut();
+
+    let blocksHeight = blocks_height();
```

Comment:
We use `snake_case` for variables in Rust: https://rust-lang.github.io/api-guidelines/naming.html